### PR TITLE
Add failing unit test for mode attribute dropping

### PR DESCRIPTION
**Pull request by niekraaijmakers**
Currently the mode attribute on the filter root get's dropped by the CPConverter.

To make sure this regression never occurs again in the future, we are adding this failing unit test.

### CODE REVIEW

**Review by anchela — CHANGES_REQUESTED** [review]
hi @niekraaijmakers , you need to have a SLING jira ticket for this. all modifications (maybe except for trivial typos) must be backed by ticket..... otherwise the release information is not accurate.

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 1]
```diff
@@ -161,15 +163,23 @@ public void convertContentPackage() throws Exception {
...
             verifyFeatureFile(outputDirectory,
                             "asd.retail.all.json",
                             "asd.sample:asd.retail.all:slingosgifeature:0.0.1",
-                            Collections.singletonList("org.apache.felix:org.apache.felix.framework:6.0.1"),
-                            Collections.singletonList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
-                            Arrays.asList("asd.sample:asd.retail.apps:zip:cp2fm-converted:0.0.1",
-                                            "asd.sample:Asd.Retail.ui.content:zip:cp2fm-converted:0.0.1",
-                                            "asd:Asd.Retail.config:zip:cp2fm-converted:0.0.1",
-                                            "asd.sample:asd.retail.all:zip:cp2fm-converted:0.0.1"));
+                            Arrays.asList("org.apache.felix:org.apache.felix.framework:6.0.1"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:179]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 2]
```diff
@@ -161,15 +163,23 @@ public void convertContentPackage() throws Exception {
...
+                            Arrays.asList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:180]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 3]
```diff
@@ -245,9 +255,11 @@ public void convertContentPackageDropContentTypePackagePolicy() throws Exception
             verifyFeatureFile(outputDirectory,
                             "asd.retail.all.json",
                             "asd.sample:asd.retail.all:slingosgifeature:0.0.1",
-                            Collections.singletonList("org.apache.felix:org.apache.felix.framework:6.0.1"),
-                            Collections.singletonList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
-                            Arrays.asList("asd.sample:asd.retail.apps:zip:cp2fm-converted:0.0.1", "asd:Asd.Retail.config:zip:cp2fm-converted:0.0.1"));
+                            Arrays.asList("org.apache.felix:org.apache.felix.framework:6.0.1"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:258]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 4]
```diff
@@ -245,9 +255,11 @@ public void convertContentPackageDropContentTypePackagePolicy() throws Exception
...
+                            Arrays.asList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:259]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 5]
```diff
@@ -316,9 +328,9 @@ public void convertContentPackagePutInDedicatedFolderContentTypePackagePolicy()
             verifyFeatureFile(outputDirectory,
                             "asd.retail.all.json",
                             "asd.sample:asd.retail.all:slingosgifeature:0.0.1",
-                            Collections.singletonList("org.apache.felix:org.apache.felix.framework:6.0.1"),
-                            Collections.singletonList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
-                            Arrays.asList("asd.sample:asd.retail.apps:zip:cp2fm-converted:0.0.1",
+                            Arrays.asList("org.apache.felix:org.apache.felix.framework:6.0.1"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:331]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java [hunk 6]
```diff
@@ -316,9 +328,9 @@ public void convertContentPackagePutInDedicatedFolderContentTypePackagePolicy()
...
+                            Arrays.asList("org.apache.sling.commons.log.LogManager.factory.config~asd-retail"),
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ContentPackage2FeatureModelConverterTest.java:332]: changing from Collections.singletonList to Arrays.asList seems to be an unrelated change. please revert if it is not required for the test to pass

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 7]
```diff
@@ -127,7 +130,7 @@
 
     public ConverterUserAndPermissionTest(@NotNull String systemUserRelPath, @Nullable String enforcePrincipalBasedSupportedPath, @NotNull String name) {
         this.aclManager = new DefaultAclManager(enforcePrincipalBasedSupportedPath, systemUserRelPath);
-        this.handlersManager = new DefaultEntryHandlersManager(Collections.emptyMap(), false, 
+        this.handlersManager = new DefaultEntryHandlersManager(Collections.emptyMap(), false,
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:133]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 8]
```diff
@@ -141,7 +144,7 @@ public void setUp() throws IOException {
 
         outputDirectory = new File(System.getProperty("java.io.tmpdir"), getClass().getName() + '_' + System.currentTimeMillis());
         featuresManager = new DefaultFeaturesManager(true, 5, outputDirectory, null, null, null, aclManager);
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:147]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 9]
```diff
@@ -174,12 +177,12 @@ public void testConvertPackageWithUsersGroupsAndServiceUsers() throws Exception
     }
 
     /**
-     * "demo-cp3.zip" contains the same content as "demo-cp.zip" but with altered order leading to ACE being read before 
+     * "demo-cp3.zip" contains the same content as "demo-cp.zip" but with altered order leading to ACE being read before
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:180]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 10]
```diff
@@ -174,12 +177,12 @@ public void testConvertPackageWithUsersGroupsAndServiceUsers() throws Exception
...
      * the corresponding system-user, whose principal is referenced in the ACE.
-     * 
-     * This test would fail if user/group information was not collected during the first pass (i.e. corresponding 
+     *
+     * This test would fail if user/group information was not collected during the first pass (i.e. corresponding
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:183]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 11]
```diff
@@ -174,12 +177,12 @@ public void testConvertPackageWithUsersGroupsAndServiceUsers() throws Exception
...
      * handlers listed in {@link org.apache.sling.feature.cpconverter.vltpkg.RecollectorVaultPackageScanner}.
-     * 
+     *
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:185]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 12]
```diff
@@ -225,14 +228,43 @@ public void testConvertCONTENTPackageWithUsersGroupsAndServiceUsers() throws Exc
...
+
     private static void assertExpectedPolicies(@NotNull File converted ) throws IOException {
         assertPolicy(converted, "jcr_root/demo-cp/_rep_policy.xml", "cp-serviceuser-1", "cp-user1", "cp-group1");
         assertPolicy(converted, "jcr_root/home/groups/demo-cp/EsYrXeBdSRkna2kqbxjl/_rep_policy.xml", null, "cp-group1");
         assertPolicy(converted,  "jcr_root/home/users/demo-cp/XPXhA_RKMFRKNO8ViIhn/_rep_policy.xml", null, "cp-user1");
         assertPolicy(converted, "jcr_root/home/groups/demo-cp/_rep_policy.xml", "cp-serviceuser-3");
         assertPolicy(converted, "jcr_root/home/users/demo-cp/_rep_policy.xml", "cp-serviceuser-3");
     }
-    
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:267]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 13]
```diff
@@ -254,18 +286,18 @@ private static void assertPolicy(@NotNull File contentPackage, @NotNull String p
             }
         }
     }
-    
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:289]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 14]
```diff
@@ -254,18 +286,18 @@ private static void assertPolicy(@NotNull File contentPackage, @NotNull String p
...
     private void verifyRepoInit() throws RepoInitParsingException {
         Feature f = featuresManager.getTargetFeature();
         Extension ext = f.getExtensions().getByName(Extension.EXTENSION_NAME_REPOINIT);
         assertNotNull(ext);
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:294]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 15]
```diff
@@ -254,18 +286,18 @@ private static void assertPolicy(@NotNull File contentPackage, @NotNull String p
...
         List<Operation> ops = new RepoInitParserService().parse(new StringReader(ext.getText()));
         int size = (enforcePrincipalBased) ? 7 : 8;
         assertEquals(size, ops.size());
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:298]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 16]
```diff
@@ -254,18 +286,18 @@ private static void assertPolicy(@NotNull File contentPackage, @NotNull String p
...
         assertEquals(1, ops.stream().filter(operation -> operation instanceof RegisterNodetypes).count());
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:300]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 17]
```diff
@@ -277,7 +309,7 @@ private void verifyRepoInit() throws RepoInitParsingException {
             assertFalse(createServiceUsers.stream().anyMatch(operation -> ((CreateServiceUser) operation).isForcedPath()));
             assertEquals(2, createServiceUsers.stream().filter(operation -> ((CreateServiceUser) operation).getPath().startsWith(withRelPath+"/demo-cp")).count());
         }
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:312]: unrelated change. can be omitted to simplify reading the diff.

File: src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java [hunk 18]
```diff
@@ -288,7 +320,7 @@ private void verifyRepoInit() throws RepoInitParsingException {
             assertEquals(1, setAclPrincipalBased.size());
         }
         assertTrue(ops.stream().noneMatch(op -> op instanceof SetAclPaths));
-        
+
```
Comment (anchela) [inline src/test/java/org/apache/sling/feature/cpconverter/ConverterUserAndPermissionTest.java:323]: unrelated change. can be omitted to simplify reading the diff.

**Review by anchela — CHANGES_REQUESTED** [review]

**Comment by karlpauls** [comment]
Should be covered by #98 now.

**Pull request closed without merging**